File: OJ/usaco/cpp/palsquare.cpp
/*
ID:oifox20071
LANG:C++
PROG:palsquare
*/
#include <iostream>
using namespace std;
const int n = 300;
int b;

bool check(string temp)
{
	for (int i = 0; i < temp.size() >> 1; ++ i)
		if (temp[i] != temp[temp.size() - i - 1]) return false;
	return true;
}

char digit(int k)
{
	return (k > 9) ? char(55 + k) : char(48 + k);
}

string change(int x)
{
	string temp = "";
	for (; x > 0; x /= b) temp = digit(x % b) + temp;
	return temp;
}

int main()
{
	freopen("palsquare.in", "r", stdin); freopen("palsquare.out", "w", stdout);
	cin >> b;
	for (int i = 1; i <= n; ++ i)
	{
		string temp = change(i * i);
		if (check(temp))
		{
			cout << change(i) << ' '<< temp << endl;
		}
	}
	return 0;
}

File: OJ/usaco/pascal/s1/s12/palsquare.pas
{
ID:lijianx1
PROG:palsquare
LANG:PASCAL
}
program palsquare;
var s:string;
    i,n:longint;
function check(a:string):boolean;
 var l,i:longint;
 begin
     l:=length(a);
     for i:=1 to l div 2 do 
	if a[i]<>a[l-i+1] then exit(false);
     check:=true;
 end;

function digit(k:longint):char;
 begin
     if k>9 then digit:=chr(55+k) else digit:=chr(48+k);
 end;

function change(x:longint):string;
 begin
     change:='';
     while x>0 do begin
	 change:=digit(x mod n)+change;
	 x:=x div n;
     end;
 end;

begin
    assign(input,'palsquare.in'); reset(input);
    assign(output,'palsquare.out'); rewrite(output);
    readln(n);
    for i:=1 to 300 do begin
	s:=change(i*i);
	if check(s) then writeln(change(i),' ',s);
    end;
    close(input);
    close(output);
end.